Author: Agustin Martin Domingo <agmartin@debian.org>
Description: make some definitions also work with XEmacs.

--- a/support/emacsen/ispell.el
+++ b/support/emacsen/ispell.el
@@ -3254,8 +3254,10 @@ ispell-region: Search for first region t
                      "ispell-region: First skip: %s at (pos,line,column): (%s,%s,%s).\n"
                      key
                      (save-excursion (goto-char skip-region-start) (point))
-                     (line-number-at-pos skip-region-start)
-                     (save-excursion (goto-char skip-region-start) (current-column)))))
+		     (if (featurep 'xemacs)
+			 (line-number skip-region-start)
+		       (line-number-at-pos skip-region-start))
+		     (save-excursion (goto-char skip-region-start) (current-column)))))
 	      (ispell-print-if-debug
                "ispell-region: Continue spell-checking with %s and %s dictionary...\n"
 		program-basename dictionary)
@@ -3296,7 +3298,9 @@ ispell-region: Search for first region t
                              "ispell-region: Next skip: %s at (pos,line,column): (%s,%s,%s).\n"
                              key
                              (save-excursion (goto-char skip-region-start) (point))
-                             (line-number-at-pos skip-region-start)
+			     (if (featurep 'xemacs)
+				 (line-number skip-region-start)
+			       (line-number-at-pos skip-region-start))
                              (save-excursion (goto-char skip-region-start) (current-column))))
 			(set-marker skip-region-start nil))))
 		(setq reg-end (max (point)
@@ -3604,7 +3608,10 @@ Returns the sum SHIFT due to changes in
 		;; This error occurs due to filter pipe problems
 		(let* ((ispell-pipe-word (car poss))
 		       (actual-point (marker-position word-start))
-		       (actual-line (line-number-at-pos actual-point))
+		       (actual-line
+			(if (featurep 'xemacs)
+			    (line-number actual-point)
+			  (line-number-at-pos actual-point)))
 		       (actual-column (save-excursion (goto-char actual-point)
                                                       (current-column))))
 		  (ispell-print-if-debug
