Author: Agustin Martin Domingo <agmartin@debian.org>
Description: Minor change in ispell pop-up menu

--- a/support/emacsen/ispell.el
+++ b/support/emacsen/ispell.el
@@ -2253,7 +2253,7 @@ Global `ispell-quit' set to start locati
       (erase-buffer)
       (if guess
 	  (progn
-	    (insert "Affix rules generate and capitalize "
+	    (insert "Note: Affix rules generate and capitalize "
 		    "this word as shown below:\n\t")
 	    (while guess
 	      (if (> (+ 4 (current-column) (length (car guess)))
