Author: Agustin Martin Domingo <agmartin@debian.org>
Description: Protect agaiinst defface definition syntax not available until 24.4.

--- a/support/emacsen/flyspell.el
+++ b/support/emacsen/flyspell.el
@@ -447,13 +447,17 @@ like <img alt=\"Some thing.\">."
 ;;*---------------------------------------------------------------------*/
 ;;*    Highlighting                                                     */
 ;;*---------------------------------------------------------------------*/
+(let ((may-support-wave (ispell-check-minver "24.4" emacs-version)))
 (defface flyspell-incorrect
   (if (featurep 'xemacs)
       '((((class color)) (:foreground "OrangeRed" :bold t :underline t))
 	(t (:bold t)))
-    '((((supports :underline (:style wave)))
-       :underline (:style wave :color "Red1"))
-      (t :underline t :inherit error)))
+    (if may-support-wave
+	'((((supports :underline (:style wave)))
+	   :underline (:style wave :color "Red1"))
+	  (t :underline t :inherit error))
+      '((((class color)) (:foreground "OrangeRed" :bold t :underline t))
+	(t (:bold t)))))
   "Flyspell face for misspelled words."
   :version "24.4"
   :group 'flyspell)
@@ -462,13 +466,18 @@ like <img alt=\"Some thing.\">."
   (if (featurep 'xemacs)
       '((((class color)) (:foreground "Gold3" :bold t :underline t))
 	(t (:bold t)))
-    '((((supports :underline (:style wave)))
-       :underline (:style wave :color "DarkOrange"))
-      (t :underline t :inherit warning)))
+    (if may-support-wave
+	'((((supports :underline (:style wave)))
+	   :underline (:style wave :color "DarkOrange"))
+	  (t :underline t :inherit warning))
+      '((((class color)) (:foreground "Gold3" :bold t :underline t))
+	(t (:bold t)))))
   "Flyspell face for words that appear twice in a row.
 See also `flyspell-duplicate-distance'."
   :version "24.4"
   :group 'flyspell)
+) ;; End of let
+
 
 (defvar flyspell-overlay nil)
 
